internal/cleanup/cleanup_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package cleanup
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/scheduler"
+	"github.com/hashicorp/boundary/internal/session"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPrototypeInsertDeleteAndClear(t *testing.T) {
+	ctx := context.Background()
+
+	conn, _ := db.TestSetup(t, "postgres")
+	wrapper := db.TestWrapper(t)
+	iamRepo := iam.TestRepo(t, conn, wrapper)
+	session := session.TestDefaultSession(t, conn, wrapper, iamRepo)
+	rw := db.New(conn)
+
+	db, err := conn.SqlDB(ctx)
+	if err != nil {
+		t.Errorf("error getting db connection %s", err)
+	}
+
+	_, err = db.Exec(fmt.Sprintf(`delete from target where public_id = '%s'`, session.TargetId))
+	if err != nil {
+		t.Errorf("error deleting from target %s", err)
+	}
+
+	// make sure that the trigger works
+	rows, err := db.Query("select count(public_id) from target_deleted")
```

Comment:
Can we use `QueryRowContext`?
```suggestion
	var count int
	err := db.QueryRowContext(ctx, "select count(public_id) from target_deleted").Scan(&count)
```